tools/tflitefile_tool/select_operator.py

```diff
@@ -1120,30 +1213,49 @@ def main(args):
     sample_model = tflite.Model.Model.GetRootAsModel(sample_buf, 0)
     sample_subgraph = sample_model.Subgraphs(subgraph)
 
+    used_subgraphs_list = []
+    used_subgraphs_list.append(args.subgraph)
+    GetUsedSubgraphsList(sample_model, args.subgraph, operator_list, used_subgraphs_list)
+
+    used_subgraphs_dic = {}
+    for new_subgraph_idx in range(len(used_subgraphs_list)):
+        sample_subgraph_idx = used_subgraphs_list[new_subgraph_idx]
+        used_subgraphs_dic[sample_subgraph_idx] = new_subgraph_idx
+
     # Collect used tensor & used operator
     used_tensors = []
-    used_operators = []
+    used_opcodes = []
 
-    for opcode_idx in opcode_list:
-        opcode = sample_subgraph.Operators(opcode_idx)
-        for input_idx in range(opcode.InputsLength()):
-            input_tensor_idx = opcode.Inputs(input_idx)
+    for operator_idx in operator_list:
+        operator = sample_subgraph.Operators(operator_idx)
+        for input_idx in range(operator.InputsLength()):
+            input_tensor_idx = operator.Inputs(input_idx)
             if not input_tensor_idx == -1 and not input_tensor_idx in used_tensors:
                 # default: same as input sample
                 used_tensors.append(input_tensor_idx)
 
-        for output_idx in range(opcode.OutputsLength()):
-            output_tensor_idx = opcode.Outputs(output_idx)
+        for output_idx in range(operator.OutputsLength()):
+            output_tensor_idx = operator.Outputs(output_idx)
             if not output_tensor_idx in used_tensors:
                 # default: same as input sample
                 used_tensors.append(output_tensor_idx)
```

Comment:
@ragmani I think `used_tensors` could be updated with the tensors used in subgraphs from `If` and `While`.

Reply:
`used_tensors` just has tensors a selected subgraph.

Reply:
@ragmani You mean that it is guaranteed that no additional `used_tensors` will be used except for those by _selected operators of a selected subgraph_? If so, it is okay.

Reply:
(ADDED) Those updated buffers of subgraphs to `used_buffers` are not calculated from `used_tensors`. The buffers are not related to tensors of `used_tensors`.

Reply:
@glistening 
The code is to add buffers of tensors of selected subgraphs. But missing buffers were in other child subgraphs branched by control flow operations. So it was necessary to add the buffers of the child subgraphs to `used_buffers`. And I did it like below
```python
    # Append buffers of tensors of child subgraphs
    for subgraph_idx in used_subgraphs_list:
        if subgraph_idx == subgraph:
            continue
        for tensor_idx in range(sample_model.Subgraphs(subgraph_idx).TensorsLength()):
            tensor = sample_model.Subgraphs(subgraph_idx).Tensors(tensor_idx)
            used_buffers.append(tensor.Buffer())
```
The buffers are not related to tensors of `used_tensors`.

Reply:
@ragmani 

> The code is to add buffers of tensors of selected subgraph. But missing buffers were in other child subgraphs branched by control flow operations

Definitely I know. It is why I wrote 

> If you are going to update used_buffer, it is okay.